go/test/endtoend/encryption/encryptedtransport/encrypted_transport_test.go

```diff
@@ -0,0 +1,456 @@
+	qr, err = vc.Execute(ctx, request)
+	err = vterrors.FromVTRPC(qr.Error)
+	assert.NotNil(t, err)
+	assert.Contains(t, err.Error(), "table acl error")
+	assert.Contains(t, err.Error(), "cannot run PASS_SELECT on table")
+
+	// 'vtgate client 1' is authorized to access vt_insert_test
+	callerID := &vtrpc.CallerID{
+		Principal: "vtgate client 1",
+	}
+	request = getRequestWithCallerID(callerID, "select * from vt_insert_test")
+	qr, err = vc.Execute(ctx, request)
+	err = vterrors.FromVTRPC(qr.Error)
+	assert.Nil(t, err)
+
+	// 'vtgate client 2' is not authorized to access vt_insert_test
+	callerID = &vtrpc.CallerID{
+		Principal: "vtgate client 2",
+	}
+	request = getRequestWithCallerID(callerID, "select * from vt_insert_test")
+	qr, err = vc.Execute(ctx, request)
+	err = vterrors.FromVTRPC(qr.Error)
+	assert.NotNil(t, err)
+	assert.Contains(t, err.Error(), "table acl error")
+	assert.Contains(t, err.Error(), "cannot run PASS_SELECT on table")
+
+	clusterInstance.Teardown()
+}
+
+func clusterSetUp(t *testing.T) (int, error) {
+	var mysqlProcesses []*exec.Cmd
+	clusterInstance = cluster.NewCluster(cell, hostname)
+
+	// Start topo server
+	if err := clusterInstance.StartTopo(); err != nil {
+		return 1, err
+	}
+
+	// create all certs
+	log.Info("Creating certificates")
+	certDirectory = path.Join(clusterInstance.TmpDirectory, "certs")
+	_ = encryption.CreateDirectory(certDirectory, 0700)
+
+	err := encryption.ExecuteVttlstestCommand("-root", certDirectory, "CreateCA")
+	assert.Nil(t, err)
+
+	err = createSignedCert("ca", "01", "vttablet-server", "vttablet server CA")
+	assert.Nil(t, err)
+
+	err = createSignedCert("ca", "02", "vttablet-client", "vttablet client CA")
+	assert.Nil(t, err)
+
+	err = createSignedCert("ca", "03", "vtgate-server", "vtgate server CA")
+	assert.Nil(t, err)
+
+	err = createSignedCert("ca", "04", "vtgate-client", "vtgate client CA")
+	assert.Nil(t, err)
+
+	err = createSignedCert("vttablet-server", "01", "vttablet-server-instance", "vttablet server instance")
+	assert.Nil(t, err)
+
+	err = createSignedCert("vttablet-client", "01", "vttablet-client-1", "vttablet client 1")
+	assert.Nil(t, err)
+
+	err = createSignedCert("vtgate-server", "01", "vtgate-server-instance", "localhost")
+	assert.Nil(t, err)
+
+	err = createSignedCert("vtgate-client", "01", "vtgate-client-1", "vtgate client 1")
+	assert.Nil(t, err)
+
+	err = createSignedCert("vtgate-client", "02", "vtgate-client-2", "vtgate client 2")
+	assert.Nil(t, err)
+
+	for _, keyspaceStr := range []string{keyspace} {
+		KeyspacePtr := &cluster.Keyspace{Name: keyspaceStr}
+		keyspace := *KeyspacePtr
+		if err := clusterInstance.VtctlProcess.CreateKeyspace(keyspace.Name); err != nil {
+			return 1, err
+		}
+		shard := &cluster.Shard{
+			Name: shardName,
+		}
+		for i := 0; i < 2; i++ {
+			// instantiate vttablet object with reserved ports
+			tabletUID := clusterInstance.GetAndReserveTabletUID()
+			tablet := clusterInstance.GetVttabletInstance("replica", tabletUID, cell)
```

Comment:
nit: if you can pass tabletUID=0 then you don't need to call separate `tabletUID := clusterInstance.GetAndReserveTabletUID()`

Reply:
done